aws/resource_aws_gamelift_queue_test.go

```diff
@@ -0,0 +1,264 @@
+						fmt.Sprintf("%d", *uPlayerLatencyPolicies.MaximumIndividualPlayerLatencyMilliseconds)),
+
+					resource.TestCheckResourceAttr("aws_gamelift_queue.test",
+						"player_latency_policies.0.policy_duration_seconds",
+						fmt.Sprintf("%d", *uPlayerLatencyPolicies.PolicyDurationSeconds)),
+
+					resource.TestCheckResourceAttr("aws_gamelift_queue.test",
+						"timeout_in_seconds", fmt.Sprintf("%d", uTimeoutInSeconds)),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAWSGameliftQueueExists(n string, res *gamelift.GameSessionQueue) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("no Gamelift Session Queue Name is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).gameliftconn
+
+		name := rs.Primary.Attributes["name"]
+		limit := int64(1)
+		out, err := conn.DescribeGameSessionQueues(&gamelift.DescribeGameSessionQueuesInput{
+			Names: aws.StringSlice([]string{name}),
+			Limit: &limit,
+		})
+		if err != nil {
+			return err
+		}
+		attributes := out.GameSessionQueues
+		if len(attributes) < 1 {
+			return fmt.Errorf("gmelift Session Queue %q not found", name)
+		}
+		if len(attributes) != 1 {
+			return fmt.Errorf("expected exactly 1 Gamelift Session Queue, found %d under %q",
+				len(attributes), name)
+		}
+		queue := attributes[0]
+
+		if *queue.Name != name {
+			return fmt.Errorf("gamelift Session Queue not found")
+		}
+
+		*res = *queue
+
+		return nil
+	}
+}
+
+func testAccCheckAWSGameliftQueueDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).gameliftconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_gamelift_queue" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		limit := int64(1)
+		out, err := conn.DescribeGameSessionQueues(&gamelift.DescribeGameSessionQueuesInput{
+			Names: aws.StringSlice([]string{name}),
+			Limit: &limit,
+		})
+		if err != nil {
+			return err
+		}
+
+		attributes := out.GameSessionQueues
+
+		if len(attributes) > 0 {
+			return fmt.Errorf("gamelift Session Queue still exists")
+		}
+
+		return nil
+	}
+
+	return nil
+
+}
+
+func testAccAWSGameliftQueueBasicConfig(queueName string, destinations gamelift.GameSessionQueueDestination,
+	playerLatencyPolicies gamelift.PlayerLatencyPolicy, timeoutInSeconds int64) string {
+	return fmt.Sprintf(`
+resource "aws_gamelift_queue" "test" {
+  name = "%s"
+  destinations = ["%s"]
+  player_latency_policies {
+	maximum_individual_player_latency_milliseconds = %d
+	policy_duration_seconds = %d
+  }
+  timeout_in_seconds = %d
+}
+`,
+		queueName,
+		*destinations.DestinationArn,
+		*playerLatencyPolicies.MaximumIndividualPlayerLatencyMilliseconds,
+		*playerLatencyPolicies.PolicyDurationSeconds,
+		timeoutInSeconds)
+}
+
+func getComposedQueueName(name string) string {
```

Comment:
Since there is only a single usage of this very trivial function, we should remove its indirection.